src/transformers/models/llama/tokenization_llama.py

```diff
@@ -117,9 +121,33 @@ def get_vocab(self):
         vocab.update(self.added_tokens_encoder)
         return vocab
 
+    # Copied from transformers.models.t5.tokenization_t5.T5Tokenizer.tokenize
+    def tokenize(self, text, **kwargs) -> List[str]:
+        # Replace the SPIECE_UNDERLINE with a space to make sure SPIECE_UNDERLINE is only used at
+        # the beginning of the text
+        if not self.legacy:
+            text = SPIECE_UNDERLINE + text.replace(SPIECE_UNDERLINE, " ")
+        return super().tokenize(text, **kwargs)
+
+    # Copied from transformers.models.t5.tokenization_t5.T5Tokenizer._tokenize
     def _tokenize(self, text):
-        """Returns a tokenized string."""
-        return self.sp_model.encode(text, out_type=str)
+        """
+        Returns a tokenized string. Since the sentencpiece internal model always adds a SPIECE_UNDERLINE, at the
+        beginning of the provided text, we need to remove it by hand when the current text is a subsequence. This
+        happens whenever the `self.tokenize` function is called with specials tokens: the input is split on the special
+        tokens, and each subsequence is passed to `_tokenize`. Thus is a subsequence did not start with a `" "` or
+        SPIECE_UNDERLINE, we have to remove the extra `SPIECE_UNDERLINE` prepended.
```

Comment:
nit: formatting / spelling 
```suggestion
        Returns a tokenized string. 
        
        Since the sentencepiece internal model always adds a SPIECE_UNDERLINE, at the
        beginning of the provided text, we need to remove it by hand when the current text is a subsequence. This
        happens whenever the `self.tokenize` function is called with specials tokens: the input is split on the special
        tokens, and each subsequence is passed to `_tokenize`. Thus if a subsequence did not start with a `" "` or
        SPIECE_UNDERLINE, we have to remove the extra `SPIECE_UNDERLINE` prepended.
```